Modules/vc-module-pricing/VirtoCommerce.PricingModule.Web/Module.cs

```diff
@@ -131,6 +136,19 @@ public void PostInitialize(IApplicationBuilder appBuilder)
             {
                 Children = new List<IConditionTree>() { GetPricingDynamicExpression() }
             };
+
+            var registrar = _serviceProvider.GetService<IKnownExportTypesRegistrar>();
+            registrar.RegisterType<Price>()
+                .WithDataSourceFactory(dataQuery => new PriceExportPagedDataSource(_serviceProvider.GetService<IPricingSearchService>()) { DataQuery = dataQuery })
```

Comment:
Could you move pricing services resolving
```
_serviceProvider.GetService<IPricingSearchService>()
```
to variables and use them in WithDataSourceFactory?